File: Android project/FragmentClassWork/app/src/main/java/com/mirkamol/fragmentclasswork/SecondFragment.java
package com.mirkamol.fragmentclasswork;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class SecondFragment extends Fragment {
    private static final String KEY = "the_key";

   public static SecondFragment newInstance(String text){
       SecondFragment fragment = new SecondFragment();
        Bundle bundle = new Bundle();
        bundle.putString(KEY, text);
        fragment.setArguments(bundle);
        return fragment;
   }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_second, container, false);
        TextView textView = view.findViewById(R.id.tv_second_fragment);
        textView.setText(getArguments().getString(KEY));
        return view;
    }
}

File: Android project/JetpackNavigationKotlin/app/src/main/java/com/mirkamol/jetpacknavigationkotlin/ui/fragments/SecondFragment.kt
package com.mirkamol.jetpacknavigationkotlin.ui.fragments

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.navigation.fragment.findNavController
import com.mirkamol.jetpacknavigationkotlin.R
import com.mirkamol.jetpacknavigationkotlin.databinding.FragmentSecondBinding
import com.mirkamol.jetpacknavigationkotlin.utils.viewBinding

class SecondFragment : Fragment(R.layout.fragment_second) {
    private val binding by viewBinding { FragmentSecondBinding.bind(it) }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        binding.DoNotClick.setOnClickListener {
            findNavController().navigate(R.id.thirdFragment)
        }

    }

}